### PR TITLE
TypeScript definition file

### PR DESCRIPTION
**Pull request by alex-greff**
Hi,
I ended up using this library in a school project which was using TypeScript so I had to make a definition file for this library.

I figured since I already spent the time to make it that it would be nice to put it into the actual library so others can use it.

Let me know if this is something you're interested in.

### CODE REVIEW

**Comment by hvianna** [comment]
Cool, thanks! Will review and merge this ASAP.

**Comment by Staijn1** [comment]
Any update on this? Looks pretty usefull

**Comment by hvianna** [comment]
@alex-greff Would you be willing to update this PR to reflect the [changes and new features in version 3.0.0](https://github.com/hvianna/audioMotion-analyzer/releases/tag/3.0.0)?

**Comment by alex-greff** [comment]
Sure, I'll look into it.

**Comment by alex-greff** [comment]
Okay the TypeScript declarations are updated for v3.

**Comment by hvianna** [comment]
@alex-greff Awesome, thank you so much! I’ll review it over the weekend.

**Comment by alex-greff** [comment]
@hvianna I think I found a small typo in the documentation on the website
image
It seems like that should actually be `showScaleX` instead of `showScale`

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,183 @@
+type OnCanvasDrawFunction = (instance: AudioMotionAnalyzer) => unknown;
+type OnCanvasResizeFunction = (
+  reason: CanvasResizeReason,
+  instance: AudioMotionAnalyzer
+) => unknown;
+
+type CanvasResizeReason = "create" | "fschange" | "lores" | "resize" | "user";
+
+export interface Options {
+  audioCtx?: AudioContext;
```
Comment (hvianna) [inline index.d.ts:10]: `audioCtx` and `source` are only valid in the constructor call, but this interface will be used by the `setOptions()` method as well.
Can we remove them from this interface and create another one for the constructor definition, say `ConstructorOptions`, which extends this and adds those two properties?

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+  get mode(): number;
+  set mode(value: number);
+
+  public overlay: boolean;
+
+  get peakEnergy(): number;
+  get pixelRatio(): number;
+
+  get radial(): number;
+  set radial(value: number);
```
Comment (hvianna) [inline index.d.ts:128]:
```suggestion
  get radial(): boolean;
  set radial(value: boolean);
```

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+  static get version(): string;
+
+  public onCanvasDraw: OnCanvasDrawFunction | undefined;
+  public onCanvasResize: OnCanvasResizeFunction | undefined;
+
+  public connectInput(source: HTMLMediaElement): MediaElementAudioSourceNode;
+  public connectInput(source: AudioNode): AudioNode;
+  public connectOutput(node?: AudioNode): void;
+
+  public disconnectInput(node?: AudioNode): void;
```
Comment (hvianna) [inline index.d.ts:171]:
```suggestion
  public disconnectInput(node?: AudioNode | AudioNode[]): void;
```

File: index.d.ts [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+  public disconnectOutput(node?: AudioNode): void;
+
+  public registerGradient(name: string, options: GradientOptions): void;
+
+  public setCanvasSize(width: number, height: number): void;
+  public setFreqRange(minFreq: number, maxFreq: number): void;
+  public setSensitivity(minDecibels: number, maxDecibels: number): void;
+
+  public toggleFullscreen(): void;
```
Comment (hvianna) [inline index.d.ts:180]: missing [`setOptions()`](https://github.com/hvianna/audioMotion-analyzer#setoptions-options-) and [`toggleAnalyzer()`](https://github.com/hvianna/audioMotion-analyzer#toggleanalyzer-boolean-) methods

**Review by hvianna — CHANGES_REQUESTED** [review]

**Comment by hvianna** [comment]
> 
> 
> @hvianna I think I found a small typo in the documentation on the website
> image
> It seems like that should actually be `showScaleX` instead of `showScale`

Ah, true! I missed one place. Thanks for spotting this!

File: index.d.ts (hunk 4 above)
Comment (alex-greff) [reply index.d.ts:185]: Not sure how I missed those two, my bad. Updated now.

**Comment by alex-greff** [comment]
@hvianna requested changes are implemented.

**Review by hvianna — APPROVED** [review]

**Comment by Staijn1** [comment]
Awesome guys! This is really great, really appreciate it!

**Comment by hvianna** [comment]
Thanks again @alex-greff!

@Staijn1 I plan to make the release over the weekend, but you can get it already from the beta channel on npm:
```
npm i audiomotion-analyzer@beta
```

**Pull request merged**